src/ui/NotebookCell.tsx

```diff
@@ -0,0 +1,66 @@
+import { Check, CircleIcon, CornerDownRightIcon, X } from "lucide-react";
+import { Suspense, lazy } from "react";
+import { useStore } from "zustand";
+
+import CellInputFallback from "./CellInputFallback";
+import { useNotebook } from "./Notebook";
+import OutputView from "./OutputView";
+
+const CellInput = lazy(() => import("./CellInput"));
+
+type Props = {
+  id: string;
+};
+
+export default function NotebookCell({ id }: Props) {
+  const notebook = useNotebook();
+  const cell = useStore(notebook.store, (state) => state.cells[id]);
+
+  if (!cell) {
+    return null;
+  }
+
+  return (
+    <div className="flex">
+      <div className="flex-1">
+        <div className="flex items-start gap-2 px-4">
+          <CircleIcon className="mt-1.5 size-3.5 fill-gray-200 stroke-none" />
+          <div className="flex-1">
+            <Suspense fallback={<CellInputFallback cellId={id} />}>
+              <CellInput cellId={id} />
+            </Suspense>
+          </div>
+        </div>
+
+        {cell?.output && (
+          <div className="flex gap-3 px-4 pt-4">
+            <div>
+              <CornerDownRightIcon className="ml-1 size-5 text-gray-500" />
```

Comment:
note that I changed this to gray because results don't necessarily imply success (e.g. we write the error to `output`)